perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -222,6 +222,22 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
 		return attributes;
 	}
 
+	public List<Attribute> getAttributes(PerunSession sess, Resource resource, Member member, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException {
+		this.checkMemberIsFromTheSameVoLikeResource(sess, member, resource);
+		if(attrNames.isEmpty()) return this.getAttributes(sess, resource, member, workWithUserAttributes);
+
+		List<Attribute> attributes = new ArrayList<>();
```

Comment:
I didn't check, but do all getAttributes(attrNames) methods silently skip wrong urns/namespaces ? Since you use same list of names for all namespaces.